chainstate/src/detail/tests/normal_tokens.rs

```diff
@@ -0,0 +1,1650 @@
+                        }),
+                        OutputPurpose::Transfer(Destination::AnyoneCanSpend),
+                    ))
+                    .build(),
+            )
+            .build_and_process();
+
+        assert!(matches!(
+            result,
+            Err(BlockError::CheckBlockFailed(
+                CheckBlockError::CheckTransactionFailed(CheckBlockTransactionsError::TokensError(
+                    TokensError::IssueErrorInvalidTickerLength(_, _)
+                ))
+            ))
+        ));
+
+        // Ticker doesn't exist
+        let result = tf
+            .make_block_builder()
+            .add_transaction(
+                TransactionBuilder::new()
+                    .add_input(TxInput::new(
+                        outpoint_source_id.clone(),
+                        0,
+                        InputWitness::NoSignature(None),
+                    ))
+                    .add_output(TxOutput::new(
+                        OutputValue::Token(TokenData::TokenIssuanceV1 {
+                            token_ticker: b"".to_vec(),
+                            amount_to_issue: Amount::from_atoms(rng.gen_range(1..u128::MAX)),
+                            number_of_decimals: rng.gen_range(1..18),
+                            metadata_uri: random_string(&mut rng, 1..1024).as_bytes().to_vec(),
+                        }),
+                        OutputPurpose::Transfer(Destination::AnyoneCanSpend),
+                    ))
+                    .build(),
+            )
+            .build_and_process();
+
+        assert!(matches!(
+            result,
+            Err(BlockError::CheckBlockFailed(
+                CheckBlockError::CheckTransactionFailed(CheckBlockTransactionsError::TokensError(
+                    TokensError::IssueErrorInvalidTickerLength(_, _)
+                ))
+            ))
+        ));
+
+        // Ticker contain non alpha-numeric char
+        let result = tf
+            .make_block_builder()
+            .add_transaction(
+                TransactionBuilder::new()
+                    .add_input(TxInput::new(
+                        outpoint_source_id.clone(),
+                        0,
+                        InputWitness::NoSignature(None),
+                    ))
+                    .add_output(TxOutput::new(
+                        OutputValue::Token(TokenData::TokenIssuanceV1 {
+                            token_ticker: "💖".as_bytes().to_vec(),
+                            amount_to_issue: Amount::from_atoms(rng.gen_range(1..u128::MAX)),
+                            number_of_decimals: rng.gen_range(1..18),
+                            metadata_uri: random_string(&mut rng, 1..1024).as_bytes().to_vec(),
+                        }),
+                        OutputPurpose::Transfer(Destination::AnyoneCanSpend),
+                    ))
+                    .build(),
+            )
+            .build_and_process();
+
+        assert!(matches!(
```

Comment:
I have doubts. `assert_eq!`  require use real parameters instead of `_`. These parameters might be random or easily changed due to changes in a test.